api/v1alpha1/kvmcluster_types.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+// KVMClusterSpec defines the desired state of KVMCluster
+type KVMClusterSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// CalicoNetwork defines the network configuration to be used for Calico CNI.
+	// +optional
+	CalicoNetwork CalicoNetwork `json:"calicoNetwork"`
+
+	// DockerNetwork defines the network configuration to be used for the docker daemon.
+	// +optional
+	DockerNetwork DockerNetwork `json:"dockerNetwork"`
```

Comment:
This could also go in the KVMMachine, but I assume it would be the same for all machines in a cluster.

Reply:
Might drop this if we are targeting containerd only and not docker.

Reply:
I _think_ this belongs to the KVMMachine and if I got that right, we can use a KVMMachineTemplate that contains the repetitive data.

Reply:
The docs are pretty sparse around MachineDeployments, but based on https://tanzu.vmware.com/content/blog/taking-kubernetes-to-the-people-how-cluster-api-promotes-self-service-infrastructure, the `KVMMachineTemplate` makes sense. I'll drop this field here.

Reply:
I added a `KVMMachineTemplate` type which allows any field from the `KVMMachineSpec` to be templated. This also allows us to use `MachineDeployment` which is nice.